ls1intum/Artemis
# `Development`: Fix flaky modeling assessment component tests

---

> **Warning**
> **Before merging:**
> - [X] This PR needs to be merged in Apollon: https://github.com/ls1intum/Apollon/pull/296
> - [X] The Apollon package needs to be updated to `@ls1intum/Apollon 2.<newversion>` and *all* references to `@palsch/apollon` need to be removed.

---

### Checklist
#### General



#### Client

### Motivation and Context


Some modeling assessment tests are currently flaky:
- Bamboo: https://bamboo.ase.in.tum.de/build/result/viewBuildResults.action?buildKey=ARTEMIS-TESTS5145-TSTEST&buildNumber=4
- Github: Failing more consistently: https://github.com/ls1intum/Artemis/actions/runs/5547057107/job/15030418353?pr=6892

### Description

Used Angular change detection with `whenStable` to wait for the editor to be properly initialized. This still was not enough, as the React render cycle still does not line up with the one from Angular. Therefore, I introduced a new promise to wait for the next React render cycle in Apollon (PR https://github.com/ls1intum/Apollon/pull/296), and I'm waiting for that one now. This also allowed me to remove many test and production code delays.

For testing (locally, on a test server, running the client tests), I changed 

### Steps for Testing

Generally test modeling exercise assessment again and try to be creative:
- Assess a normal modeling exercise or one in an exam
- Make sure that the correct elements are highlighted in the assessment editor at all times
- Make sure that feedbacks on all kinds of elements still work and can be saved, also on Safari if possible (see also https://github.com/ls1intum/Apollon/pull/295)

### Code Review
Ignore all replacements `@ls1intum/apollon` -> `@palsch/apollon`. They are only for testing and will be removed.

### Review Progress






#### Manual Tests

#### Code Review

### Test Coverage






Unchanged.
